core/controllers/collection_viewer_test.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2015 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for the page that allows learners to play through a collection."""
+
+__author__ = 'Ben Henning'
+
+from core.controllers import collection_viewer
+from core.domain import collection_domain
+from core.domain import collection_services
+from core.domain import rights_manager
+from core.tests import test_utils
+import feconf
+
+
+class CollectionViewerPermissionsTest(test_utils.GenericTestBase):
+    """Test permissions for learners to view collections."""
+
+    COL_ID = 'cid'
+
+    def setUp(self):
+        """Before each individual test, create a dummy collection."""
+        super(CollectionViewerPermissionsTest, self).setUp()
+
+        self.signup(self.EDITOR_EMAIL, self.EDITOR_USERNAME)
+        self.EDITOR_ID = self.get_user_id_from_email(self.EDITOR_EMAIL)
+
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.VIEWER_ID = self.get_user_id_from_email(self.VIEWER_EMAIL)
+
+        self.save_new_valid_collection(self.COL_ID, self.EDITOR_ID)
+
+    def test_unpublished_collections_are_invisible_to_logged_out_users(self):
+        response = self.testapp.get(
+            '%s/%s' % (feconf.COLLECTION_URL_PREFIX, self.COL_ID),
+            expect_errors=True)
+        self.assertEqual(response.status_int, 404)
+
+    def test_unpublished_collections_are_invisible_to_unconnected_users(self):
+        self.login(self.VIEWER_EMAIL)
```

Comment:
I wonder if it might be better to have a different name for this user with different semantics -- maybe NEW_USER or just USER. The reason I mention this is because VIEWER sometimes connotes "someone who has permissions to view a particular exploration". Fixing this here is optional, though, since we already use VIEWER inconsistently.